clients/admin-ui/src/features/system/AddMultipleSystems.tsx

```diff
@@ -0,0 +1,249 @@
+  <Flex alignItems="center" height="100%">
+    <Badge>{value.split(".")[0] === "gvl" ? "GVL" : "AC"}</Badge>
+  </Flex>
+);
+
+type MultipleSystemTable = DictSystems;
+
+const columnHelper = createColumnHelper<MultipleSystemTable>();
+
+type Props = {
+  isSystem: boolean;
+  redirectRoute: string;
+};
+
+export const AddMultipleSystems = ({ redirectRoute, isSystem }: Props) => {
+  const systemText = isSystem ? "System" : "Vendor";
+  const toast = useToast();
+  const features = useFeatures();
+  const router = useRouter();
+  const { isLoading: isGetLoading } = useGetAllCreatedSystemsQuery(undefined, {
+    skip: !features.dictionaryService,
+  });
+  const [
+    postVendorIds,
+    { isLoading: isPostLoading, isSuccess: isPostSuccess },
+  ] = usePostCreatedSystemsMutation();
+
+  const dictionaryOptions = useAppSelector(selectAllDictSystems);
+  const [globalFilter, setGlobalFilter] = useState();
+  const { isOpen, onClose, onOpen } = useDisclosure();
+
+  const allRowsAdded = dictionaryOptions.every((d) => d.linked_system);
+  const columns = useMemo(
+    () => [
+      columnHelper.display({
+        id: "select",
+        header: ({ table }) => (
+          <IndeterminateCheckboxCell
+            {...{
+              checked: table.getIsAllRowsSelected(),
+              indeterminate: table.getIsSomeRowsSelected(),
+              onChange: table.getToggleAllRowsSelectedHandler(),
+              manualDisable: allRowsAdded,
+            }}
+          />
+        ),
+        cell: ({ row }) => (
+          <IndeterminateCheckboxCell
+            {...{
+              checked: row.getIsSelected(),
+              disabled: !row.getCanSelect(),
+              indeterminate: row.getIsSomeSelected(),
+              onChange: row.getToggleSelectedHandler(),
+              initialValue: row.original.linked_system,
+            }}
+          />
+        ),
+        meta: {
+          width: "55px",
+        },
+      }),
+      columnHelper.accessor((row) => row.legal_name, {
+        id: "legal_name",
+        cell: (props) => <DefaultCell value={props.getValue()} />,
+        header: (props) => <DefaultHeaderCell value={systemText} {...props} />,
+      }),
+      columnHelper.accessor((row) => row.vendor_id, {
+        id: "vendor_id",
+        cell: (props) => <VendorSourceCell value={props.getValue()} />,
+        header: (props) => <DefaultHeaderCell value="Source" {...props} />,
+        meta: {
+          width: "80px",
+        },
+      }),
+    ],
+    [allRowsAdded, systemText]
+  );
+
+  const rowSelection = useMemo(() => {
+    const innerRowSelection: Record<string, boolean> = {};
+    dictionaryOptions.forEach((ds, index) => {
+      if (ds.linked_system) {
+        innerRowSelection[index] = true;
+      }
+    });
+    return innerRowSelection;
+  }, [dictionaryOptions]);
+
+  const tableInstance = useReactTable<MultipleSystemTable>({
+    columns,
+    data: dictionaryOptions,
+    getCoreRowModel: getCoreRowModel(),
```

Comment:
hooks no longer exist in the table library. Before the order that hooks ran was dependent on the order you passed them in. 

Now the order that each step (sorting, filtering, grouping, etc) is deterministic. This is why everything is explicitly passed in on a key word.

Reply:
neat, I like that more!